lib/index.js

```diff
@@ -215,13 +203,69 @@ function lowerCaseObjectKeys(obj) {
   return newObject;
 }
 
+function validateDate(date, key) {
+  // created & expires are not required
+  if(date === undefined || date === null) {
+    return date;
+  }
+  if(date instanceof Date) {
+    const timestamp = date.getTime() / 1000;
+    if(key === 'created' || key === 'expires') {
+      return Math.floor(timestamp);
+    }
+    throw new TypeError(`Unknown date pseudo-header "${key}".`);
+  }
+  if(Number.isInteger(date)) {
+    // we assume the integer is a unix time stamp
+    return date;
+  }
+  if(typeof date === 'string') {
+    // we assume this is a unix time stamp too
+    const number = Number(date);
+    if(Number.isInteger(number)) {
+      return number;
+    }
+  }
+  throw new TypeError(
+    `"${key}" must be a UNIX timestamp or JavaScript Date.`);
+}
+
+/**
+ * Ensures date based psuedo-headers are valid timestamps and
+ *  are not invalid or expired.
+ *
+ * @param {Object} options - Options to use.
+ * @param {number} now - A unix timestamp for now.
+ *
+ * @returns {Object} Formatted options.
+*/
+function validateDateRange(options, now) {
+  // shallow copy options
+  options = {...options};
+  const possibleCreated = validateDate(options.created, 'created');
+  if(typeof possibleCreated === 'number') {
+    if(options.created > now) {
+      throw new Error(
+        'Invalid signature; the signature creation time is in the future.');
+    }
+    options.created = possibleCreated;
+  }
+  const possibleExpires = validateDate(options.expires, 'expires');
+  if(typeof possibleExpires === 'number') {
+    if(options.expires < now) {
+      throw new Error('The signature has expired.');
+    }
+    options.expires = possibleExpires;
+  }
+  return options;
+}
+
 api.createSignatureString = ({includeHeaders, requestOptions} = {}) => {
   assert.object(requestOptions, 'requestOptions');
   assert.object(requestOptions.headers, 'requestOption.headers');
   assert.string(requestOptions.method, 'requestOption.method');
   assert.string(requestOptions.url, 'requestOption.url');
   assert.optionalArrayOfString(includeHeaders, 'headers');
-
```

Comment:
Let's keep this whitespace to breakup the param validation section from other code.

Reply:
Addressed here: https://github.com/digitalbazaar/http-signature-header/pull/17/commits/909d594a20d06b0963b1ea605baad00dd8c27ab4